src/bsp/src/stm32/include/SpiHeader.h

```diff
@@ -0,0 +1,37 @@
+#ifndef SPI_HEADER_H
+#define SPI_HEADER_H
+
+#include <cstdint>
+namespace EspHeader {
+
+    struct EspSystemState {
+        uint8_t connected : 1;
+        uint8_t hasNeighborData : 1;
+        uint8_t failedCrc : 1;
+        uint8_t unused : 5;
```

Comment:
This is a bit of funky c stuff. Basically, the :1 specifies that only one bit is used on the uint8_t. This allows for bitwise assignment in a struct. In the end, all the used fields are only true or false boolean flags. However, I want the flags to be exaclty 8 bits, hence the 5 for the unused. The compiler assigns bits in order (MSB->LSB) with the according length. Another example of this could be the following.:

`c++
struct ex {
   uint8_t a:2;
   uint8_t b:3;
   uint8_t c:3;
}`

In this struct we one field of 2 bits and two field fields of 3 bits consecutively inside one byte.